apps/halide/blur/main.cpp

```diff
@@ -139,23 +122,22 @@ int exec_parrot(blurtype func, std::string output_name, int width, int height,
 
 int main() {
   const char *read_file = "gray_scaled.png";
-  std::vector<uint8_t> buffer;
+  std::vector<uint16_t> buffer;
   int width, height;
 
   if (read_png_file(read_file, buffer, width, height)) {
     printf("width: %d\n", (int)width);
     printf("height: %d\n", (int)height);
 
-    uint8_t *parrot;
-    parrot = (uint8_t *)malloc(sizeof(uint8_t) * (width + 4) * (height + 4));
+    uint16_t *parrot;
+    parrot = (uint16_t *)malloc(sizeof(uint16_t) * (width + 2) * (height + 2));
     for (int i = 0; i < height; i++) {
-      memcpy(&parrot[(i + 2) * (width + 4) + 2], &buffer[i * width],
-          sizeof(uint8_t) * width);
+      memcpy(&parrot[(i + 1) * (width + 2) + 1], &buffer[i * width],
+          sizeof(uint16_t) * width);
     }
 
```

Comment:
Can we make exec_parrot return the output buffer, then we can add a sanity check for correctness at the end of the file instead of having to run `diff` on the outputs? You can use `memcmp` or directly compare vectors if you use std::vector.